src/test/java/pl/zaradny/springApp/api/ProductEndpointTest.java

```diff
@@ -121,6 +159,60 @@ public void shouldCreateProductWithDescription(){
         assertThat(result.getBody().getDescription().getText()).isEqualTo("DescriptionDupa");
     }
 
+    @Test
+    public void shouldCreateProductWithTag(){
+        //given
+        List<TagDto> tagsList = new ArrayList<>();
+        tagsList.add(new TagDto("tag1"));
+        tagsList.add(new TagDto("tag2"));
+        final ProductRequestDto product = new ProductRequestDtoBuilder("product", new PriceDto("100", "PLN"))
+                .withTags(tagsList).build();
+        String productJson = mapToJson(product);
+        //when
+        ResponseEntity<ProductResponseDto> result = httpClient.postForEntity(productsUrl, getHttpRequest(productJson), ProductResponseDto.class);
+        //then
+        assertThat(result.getStatusCodeValue()).isEqualTo(200);
```

Comment:
tutaj jest mały artykuł z przykładem custom assercji: https://joel-costigliola.github.io/assertj/assertj-core-custom-assertions.html będziesz miał ochotę to możesz stworzyć taką dla body i lecieć fajnie chain-em ;)